extract_car_size.py

```diff
@@ -1,44 +1,106 @@
-        'Global Time',
-        'Local X',
-        'Local Y',
-        'Global X',
-        'Global Y',
-        'Vehicle Length',
-        'Vehicle Width',
-        'Vehicle Class',
-        'Vehicle Velocity',
-        'Vehicle Acceleration',
-        'Lane Identification',
-        'Preceding Vehicle',
-        'Following Vehicle',
-        'Spacing',
-        'Headway'
-    ))
+if opt.map == 'highD':
+    # Track dataframes
+    dtypes_dict = {
+        'Frame ID': np.int64,
+        'Vehicle ID': np.int64,
+        'Local X': np.float64,
+        'Local Y': np.float64,
+        'Vehicle Length': np.float64,
+        'Vehicle Width': np.float64,
+        'Vehicle Velocity X': np.float64,
+        'Vehicle Velocity Y': np.float64,
+        'Vehicle Acceleration X': np.float64,
+        'Vehicle Acceleration Y': np.float64,
+        'Front Sight Distance': np.float64,
+        'Back Sight Distance': np.float64,
+        'Spacing': np.float64,
+        'Headway': np.float64,
+        'Time to Collision': np.float64,
+        'Preceding Velocity X': np.float64,
+        'Preceding Vehicle': np.int64,
+        'Following Vehicle': np.int64,
+        'Left Preceding ID': np.int64,
+        'Left Alongside ID': np.int64,
+        'Left Following ID': np.int64,
+        'Right Preceding ID': np.int64,
+        'Right Alongside ID': np.int64,
+        'Right Following ID': np.int64,
+        'Lane Identification': np.int64,
+    }
+    for ts in time_slots:
+        df[ts] = pd.read_csv(os.path.join(path, f'{ts}_tracks.csv'),
+                             header=0,
+                             names=(
+                                 'Frame ID',
+                                 'Vehicle ID',
+                                 'Local X',
+                                 'Local Y',
+                                 'Vehicle Length',
+                                 'Vehicle Width',
+                                 'Vehicle Velocity',
+                                 'Vehicle Velocity Y',
+                                 'Vehicle Acceleration',
+                                 'Vehicle Acceleration Y',
+                                 'Front Sight Distance',
+                                 'Back Sight Distance',
+                                 'Spacing',
+                                 'Headway',
+                                 'Time to Collision',
+                                 'Preceding Velocity X',
+                                 'Preceding Vehicle',
+                                 'Following Vehicle',
+                                 'Left Preceding ID',
+                                 'Left Alongside ID',
+                                 'Left Following ID',
+                                 'Right Preceding ID',
+                                 'Right Alongside ID',
+                                 'Right Following ID',
+                                 'Lane Identification'
+                             ),
```

Comment:
Haha, I think you could have used an ordered dict, and then you could have simply dumped the keys 😜

Reply:
Good point. Do you want me to change this, or should we leave it for now?

Reply:
If you want to change it, I can wait before merging.